lighthouse-core/gather/gather-runner.js

```diff
@@ -456,14 +456,15 @@ class GatherRunner {
 
     const {emulatedFormFactor} = options.settings;
     // Whether Lighthouse was run on a mobile device (i.e. not on a desktop machine).
-    const IsMobileHost = hostUserAgent.includes('Android') || hostUserAgent.includes('Mobile');
+    const HostFormFactor = /android|mobile/i.test(hostUserAgent) ? 'mobile' : 'desktop';
```

Comment:
paul called me names (not allowed, but in his head) for wanting to use a regex here

Reply:
so you want to use a regex. now you got ~[2 problems](https://blog.codinghorror.com/regular-expressions-now-you-have-two-problems/)~ 3 problems, including name-calling.

🙃


-----------

case insensitivity is fine. CONFIDENT PROGRAMMING. WE"LL BE FINE.

- https://cs.chromium.org/chromium/src/components/cronet/android/java/src/org/chromium/net/impl/UserAgent.java?l=42&rcl=0efaa7ede88f9448f19193dc513e57d651cabe79
- https://cs.chromium.org/chromium/src/third_party/devtools-frontend/src/node_modules/useragent/lib/regexps.js?l=727&rcl=879d97b05dfe9f26ede820b0296b125d4a68e5ff

surely.